apollo-router/src/services/layers/apq.rs

```diff
@@ -89,10 +90,32 @@ async fn apq_request(
                 tracing::trace!("apq: cache insert");
                 let _ = request.context.insert("persisted_query_register", true);
                 cache.insert(redis_key(&query_hash), query).await;
+                Ok(request)
             } else {
-                tracing::warn!("apq: graphql request doesn't match provided sha256Hash");
+                tracing::debug!("apq: graphql request doesn't match provided sha256Hash");
+                let errors = vec![crate::error::Error {
+                    message: "provided sha does not match query".to_string(),
+                    locations: Default::default(),
+                    path: Default::default(),
+                    extensions: serde_json_bytes::from_value(json!({
+                          "code": "PERSISTED_QUERY_SHA_MISSMATCH",
+                          "exception": {
+                          "stacktrace": [
```

Comment:
I'm not sure if this qualifies as a `stacktrace` in this case?

Reply:
Indeed, I think I previously misread https://github.com/apollographql/apollo-server/blob/3d0d4b4b263e52bb36839221d9103f02ba1d6288/packages/integration-testsuite/src/httpServerTests.ts#L2672 and turned it into a `stacktrace`.

I'll update both errors (this one and persistedquerynotfound) to remove the stacktrace

Reply:
fixed in aec6a223